#ifndef GUI_STORE_EXPLORER_WDG_H
#define GUI_STORE_EXPLORER_WDG_H

////////////////////////////////////////////////////////////////////////////////
//
// Includes
//
#ifndef GUI_QUERY_TABLE_MODEL_H
#	include "gui_querytablemodel.h"
#endif

// UI
#include "ui_storeexplorer.h"

// Qt Includes
#include <QWidget>
////////////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////
namespace fm {
////////////////////////////////////////////////////////////////////////////////
namespace gui {
////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////
//
// class CStoreExplorer
//
class CStoreExplorer : public QWidget
{

	Q_OBJECT

public:// Constructors
	CStoreExplorer( QWidget* pwParent = nullptr );
	~CStoreExplorer() = default;

public slots:
	//
	//	Public Slots
	//
	void OnCurrentStoreChanged( QListWidgetItem * pCurrent, QListWidgetItem * pPrevious );
	void OnCurrentProductChanged( const QModelIndex & current, const QModelIndex & previous );

private:
	//
	//	Content
	//
	Ui::StoreExplorerUI ui;
	//
	CStoreContentModel* m_pContentModel;
};
////////////////////////////////////////////////////////////////////////////////




////////////////////////////////////////////////////////////////////////////////
} // namespace gui
////////////////////////////////////////////////////////////////////////////////
} // namespace fm
////////////////////////////////////////////////////////////////////////////////

#endif // GUI_STORE_EXPLORER_WDG_H